### PR TITLE
Feature/account activity

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
Added a new account_activity table to the metrics schema and the logic to populate it.

## How to test
Add/remove accounts/memberships/open dataset agreements and see their effects on the new table.  Perform activity on device that will trigger active counts to increment and see effects on the new table.

## Contributor license agreement signed?
CLA [X] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: api/account/tests/features/remove_account.feature [hunk 1]
```diff
@@ -3,7 +3,7 @@ Feature: Delete an account
...
   Scenario: Successful account deletion
     Given an account with a monthly membership
-    When the user's account is deleted
-    Then the request will be successful
-    And the membership is removed from stripe
-
+     When the user's account is deleted
+     Then the request will be successful
+      And the membership is removed from stripe
+      And the deleted account will be reflected in the account activity metrics
```
Comment (forslund) [inline api/account/tests/features/remove_account.feature:9]: Left-align

File: shared/selene/data/metric/repository/account_activity.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    def get_activity_by_date(self, activity_date: date) -> AccountActivity:
+        """Returns the account activity metrics for the given date."""
+        return self._select_one_into_dataclass(
+            dataclass=AccountActivity,
+            sql_file_name="get_account_activity_by_date.sql",
+            args=dict(activity_date=activity_date),
+        )
+
+    def delete_activity_by_date(self, activity_date: date):
```
Comment (forslund) [inline shared/selene/data/metric/repository/account_activity.py:110]: If it shouldn't be used why not prepend a `_` so it's not listed as a public method?

File: api/public/public_api/endpoints/google_stt.py [hunk 3]
```diff
@@ -47,13 +59,14 @@ def post(self):
         stt_response = self._call_google_stt()
         response = self._build_response(stt_response)
         self._write_stt_result_file(response)
+        self._update_account_activity_ts()
+        self._track_account_activity()
```
Comment (forslund) [inline api/public/public_api/endpoints/google_stt.py:63]: Will this update for empty data, which would be a sign of false activation?

File: api/public/public_api/endpoints/google_stt.py [hunk 4]
```diff
@@ -107,25 +115,30 @@ def _build_response(self, stt_response):
...
+                response = [alternative[0]["transcript"]]
         else:
             response = []
 
         return response
+
+    def _update_account_activity_ts(self):
+        self.account_repo.update_last_activity_ts(self.account.id)
+
+    def _track_account_activity(self):
```
Comment (forslund) [inline api/public/public_api/endpoints/google_stt.py:142]: This is replicated in a lot of places. Can it be moved to common parent class, maybe `PublicEndpoint`? or if that's not possible make it a function.

```python
def track_account_activity(db, account):
    account_activity_repository = AccountActivityRepository(db)
    account_activity_repository.increment_activity(account)
```

**Comment by forslund** [comment]
Also is this live on the test backend?

File: api/public/public_api/endpoints/google_stt.py (hunk 3 above)
Comment (chrisveilleux) [reply api/public/public_api/endpoints/google_stt.py:63]: Added a check for an empty response from the STT API call.

File: api/public/public_api/endpoints/google_stt.py (hunk 4 above)
Comment (chrisveilleux) [reply api/public/public_api/endpoints/google_stt.py:142]: moved this and the account activity timestamp update to a function.

**Comment by chrisveilleux** [comment]
Code review changes made.  Please review again.

File: api/public/public_api/endpoints/google_stt.py (hunk 3 above)
Comment (forslund) [reply api/public/public_api/endpoints/google_stt.py:63]: Um where is this check?

File: api/public/public_api/endpoints/google_stt.py (hunk 4 above)
Comment (forslund) [reply api/public/public_api/endpoints/google_stt.py:142]: They still seem to be here as members? Is there a commit missing?

**Comment by forslund** [comment]
My concerns seem to have been addressed. Looks good!

**Pull request merged**